File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/serialization/FatHeaderInfo.kt
package com.jetbrains.signatureverifier.serialization

import kotlinx.serialization.Serializable

@Serializable
data class FatHeaderInfo(
  var magic: UInt = 0u,
  var isBe: Boolean = false,
  var fatArchSize: UInt = 0u,
  val fatArchInfos: MutableList<FatArchInfo> = mutableListOf()
) {
  fun toByteArray() =
    magic.toByteArray() +
      fatArchSize.toByteArray(isBe) +
      fatArchInfos.fold(byteArrayOf()) { acc, it -> acc + it.toByteArray(isBe) }
}

File: net/JetBrains.FormatRipper/src/MachO/FatHeaderInfo.cs
using System.Collections.Generic;
using JetBrains.FormatRipper.Impl;
using JetBrains.FormatRipper.MachO.Impl;

namespace JetBrains.FormatRipper.MachO;

public class FatHeaderInfo
{
  public readonly uint Magic;
  public readonly bool IsBe;
  public readonly uint FatArchSize;
  public readonly List<FatArchInfo> FatArchInfos;

  public FatHeaderInfo(uint magic, bool isBe, uint fatArchSize, List<FatArchInfo> fatArchInfos)
  {
    Magic = magic;
    IsBe = isBe;
    FatArchSize = fatArchSize;
    FatArchInfos = fatArchInfos;
  }

  public byte[] ToByteArray() => MemoryUtil.ArrayMerge(
    MemoryUtil.ToByteArray(Magic),
    MemoryUtil.ToByteArray(FatArchSize, IsBe),
    InfosToByteArray()
  );

  private byte[] InfosToByteArray()
  {
    byte[] res = new byte[0];
    foreach (var fatArchInfo in FatArchInfos)
    {
      res = MemoryUtil.ArrayMerge(res, fatArchInfo.ToByteArray(IsBe));
    }

    return res;
  }
}